runtime/gc_base/modronapi.cpp

```diff
@@ -786,6 +786,44 @@ j9gc_allocation_threshold_changed(J9VMThread *currentThread)
 #endif /* defined(J9VM_GC_THREAD_LOCAL_HEAP) || defined(J9VM_GC_SEGREGATED_HEAP) */
 }
 
+/**
+ * Set the allocation sampling interval to trigger a J9HOOK_SAMPLED_OBJECT_ALLOCATE event
+ * when the hook event J9HOOK_SAMPLED_OBJECT_ALLOCATE is registered.
+ * 
+ * Examples:
+ * 	To trigger an event whenever 4K objects have been allocated:
+ *		j9gc_set_allocation_sampling_interval(vmThread, (UDATA)4096);
+ *	To trigger an event for every object allocation:
+ *		j9gc_set_allocation_sampling_interval(vmThread, (UDATA)0);
+ * By default, the sampling interval is 512 KB.
+ * 
+ * @parm[in] vmThread The current VM Thread
+ * @parm[in] samplingInterval The allocation sampling interval.
+ */
+void 
+j9gc_set_allocation_sampling_interval(J9VMThread *vmThread, UDATA samplingInterval)
+{
+	MM_GCExtensions *ext = MM_GCExtensions::getExtensions(vmThread->javaVM);
+	Trc_MM_AllocationSampling_setInterval_Entry(vmThread, samplingInterval, ext->allocationSamplingInterval);
+	ext->allocationSamplingInterval = samplingInterval;
+	Trc_MM_AllocationSampling_setInterval_Exit(vmThread);
+}
+/**
+ * Set the flag to enable or disable allocation sampling
+ * 
+ * @parm[in] vmThread The current VM Thread
+ * @parm[in] flag true - enable allocation sampling, false otherwise
+ */
+void 
+j9gc_set_allocation_sampling(J9VMThread *vmThread, BOOLEAN flag)
+{
+	J9JavaVM *vm = vmThread->javaVM;
+	MM_GCExtensions *ext = MM_GCExtensions::getExtensions(vm);
```

Comment:
What happens if this API is called and `j9gc_set_allocation_sampling_interval` was not called?

Reply:
The sampling interval stays as default value 512KB which is larger than maximum TLH size 128KB, and `tlhActiveMaximumSize` stays as default value which is same as `tlhMaximumSize`.